### PR TITLE
chore(api): expose keyinfo options

### PR DESCRIPTION
**Pull request by andrewxhill**
replaces https://github.com/textileio/js-hub/pull/479

fixes https://github.com/textileio/js-hub/issues/478

### CODE REVIEW

File: etc/hub.api.md [hunk 1]
```diff
@@ -162,7 +162,7 @@ export class Buckets extends GrpcAuthentication {
     remove(key: string): Promise<void>;
     removePath(key: string, path: string, root?: string): Promise<void>;
     root(key: string): Promise<RootObject | undefined>;
-    static withKeyInfo(key: KeyInfo, host?: string, debug?: boolean): Promise<Buckets>;
+    static withKeyInfo(key: KeyInfo, options?: WithKeyOptions): Promise<Buckets>;
```
Comment (carsonfarmer) [inline etc/hub.api.md:165]: Excellent

File: etc/hub.api.md [hunk 2]
```diff
@@ -912,6 +918,16 @@ export function watchMailbox(api: GrpcConnection, id: string, box: 'inbox' | 'se
 // @public
 export const Where: typeof Criterion;
 
+// @public
```
Comment (carsonfarmer) [inline etc/hub.api.md:921]: Very nice

File: packages/grpc-authentication/src/index.ts [hunk 3]
```diff
@@ -4,6 +4,22 @@ import { Identity } from '@textile/crypto'
 import { UserAuth, KeyInfo } from '@textile/security'
 import { Client } from '@textile/hub-threads-client'
 
+/**
+ * Set the options for authenticating with an API key
+ */
+export interface WithKeyOptions {
```
Comment (carsonfarmer) [inline packages/grpc-authentication/src/index.ts:10]: Great

File: packages/grpc-authentication/src/index.ts [hunk 4]
```diff
@@ -91,10 +107,10 @@ export class GrpcAuthentication extends GrpcConnection {
    * }
    * ```
    */
-  static async withKeyInfo(key: KeyInfo, host = defaultHost, debug = false) {
-    const context = new Context(host)
-    await context.withKeyInfo(key)
-    return new GrpcAuthentication(context, debug)
+  static async withKeyInfo(key: KeyInfo, options: WithKeyOptions = {}) {
```
Comment (carsonfarmer) [inline packages/grpc-authentication/src/index.ts:110]: Excellent! Thanks for this.

File: packages/users/src/api/index.ts [hunk 5]
```diff
@@ -70,7 +70,7 @@ export interface InboxListOptions {
 export interface GetThreadResponseObj {
   isDB: boolean
   name: string
-  id: string
+  id: ThreadID
```
Comment (carsonfarmer) [inline packages/users/src/api/index.ts:73]: Ok

**Review by carsonfarmer — APPROVED** [review]
LGTM

**Comment by andrewxhill** [comment]
okay, should be gtg now

**Comment by carsonfarmer** [comment]
Re-running those browser tests, as I think the docker node failed for some reason.

**Review by carsonfarmer — APPROVED** [review]
👍 again

**Pull request merged**